FoodFriends/app/src/main/java/com/example/foodfriends/activities/LogInActivity.java

```diff
@@ -31,31 +32,47 @@ protected void onCreate(Bundle savedInstanceState) {
         etPassword = findViewById(R.id.etLogInPassword);
         btnLogin = findViewById(R.id.btnLogIn);
         tvSignUp = findViewById(R.id.tvSignUpLink);
-        tvSignUp.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent i = new Intent(LogInActivity.this, SignUpActivity.class);
-                startActivity(i);
+        tvSignUp.setOnClickListener(this);
+        tvSignUp.setOnClickListener(this);
+        btnLogin.setOnClickListener(this);
 
-            }
-        });
-        btnLogin.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
+    }
+
+    /**
+     * Provides different functionality depending on view clicked
+     * Either taking user to sign up page or logging the user in**/
+    @Override
+    public void onClick(View v) {
+        switch(v.getId()){
+            case R.id.tvSignUpLink:
+                to_sign_up();
+                break;
+
+            case R.id.btnLogIn:
                 String username = etUsername.getText().toString();
                 String password = etPassword.getText().toString();
                 loginUser(username, password);
-            }
-        });
+                break;
+        }
+    }
 
+    /**
+     * Fires intent to to take user to Sign Up Activity **/
+    private void to_sign_up() {
+        Intent i = new Intent(LogInActivity.this, SignUpActivity.class);
+        startActivity(i);
     }
 
+    /**
+     * Users ParseUser object method to log in current user
+     * Takes user to main activity if log in attempt is successful
+     * **/
     private void loginUser(String username, String password) {
         ParseUser.logInInBackground(username, password, new LogInCallback() {
             @Override
             public void done(ParseUser user, ParseException e) {
                 if (e != null) {
-                    Log.i(TAG, "Something went wrong :(");
+                    Toast.makeText(LogInActivity.this, e.getMessage(), Toast.LENGTH_SHORT).show();
```

Comment:
great highlighting error message to user in a pop up. Wondering if u have custom error messages that are user friendly ex. "username or password incorrect" vs a more Java oriented error message.